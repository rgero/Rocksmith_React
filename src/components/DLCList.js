import React from 'react';
import { connect } from 'react-redux';

import _ from 'lodash';

// Import React Table
import ReactTable from 'react-table';
import 'react-table/react-table.css';

export const DLCList = (props) => {
    var data = props.dlc;
    return (
        <div className='content-container'>
            <ReactTable
                data={data}
                columns={[
                    {
                    Header: 'DLC',
                    columns: [
                        {
                        Header: 'Artist Name',
                        accessor: 'artist'
                        },
                        {
                            Header: 'Song Name',
                            accessor: 'name'
                        },
                        {
                            Header: 'Lead',
                            accessor: 'leadTuning'
                        },
                        {
                            Header: 'Rhythm',
                            accessor: 'rhythmTuning'
                        },
                        {
                            Header: 'Bass',
                            accessor: 'bassTuning'
                        }
                    ]
                    }
                ]}
                defaultPageSize={10}
                className='-striped -highlight'
                />
        </div>
    )
};

const mapStateToProps = (state, props)=>{
    return {
        dlc: state
    };
}

export default connect(mapStateToProps)(DLCList);