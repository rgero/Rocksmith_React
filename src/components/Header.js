import React from 'react';
import { Link } from 'react-router-dom';


export const Header = () => (
    <header className="header">
        <div className="header__content">
            <Link className="header__title" to="/" exact="true"><h1>Name coming soon</h1></Link>
        </div>
    </header>
);

export default Header;