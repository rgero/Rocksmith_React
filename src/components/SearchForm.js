import React from 'react';

export class SearchForm extends React.Component {

    constructor(props){
        super(props);

        //Functions
        this.onSubmit = this.onSubmit.bind(this);
        this.onTextChange = this.onTextChange.bind(this);
        this.resetState = this.resetState.bind(this);

        this.state = {
            artistNames: [],
            songNames: [],
            leadTunings: [],
            rhythmTunings: [],
            bassTunings: [],
        };
    };
    
    onSubmit(e){
        e.preventDefault();
        this.props.onSubmit({
            artistNames: this.state.artistNames,
            songNames: this.state.songNames,
            leadTunings: this.state.leadTunings,
            rhythmTunings: this.state.rhythmTunings,
            bassTunings: this.state.bassTunings
        })
    }

    resetState(){
        this.setState({
            artistNames: [],
            songNames: [],
            leadTunings: [],
            rhythmTunings: [],
            bassTunings: [],
        })
    }

    onTextChange = (key) => (e) => {
        var newValue = e.target.value;
        switch(key){
            case "artistNames":
                this.setState({
                    artistNames: newValue
                });
                break;
            default:
                break;
        }
    }

    render() {
        return (
            <div className="content-container">
                <form className="form" onSubmit={this.onSubmit}>
                    <div className="form__input">
                        <label>Artist Names</label>
                        <input
                            type="text"
                            className="text-input"
                            placeholder="Names"
                            autoFocus
                            value={this.state.artistNames}
                            onChange={this.onTextChange("artistNames")}
                        />
                    </div>
                    <div className="buttons">
                        <button className="button" onClick={this.resetState}>Reset</button>
                        <button className="button">Submit</button>
                    </div>
                </form>
            </div>
        )
    }

}


export default SearchForm;