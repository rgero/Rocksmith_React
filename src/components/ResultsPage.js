import React, { Component } from 'react';

//Components
import Header from './Header';
import DLCList from './DLCList';

export class ResultsPage extends Component {
  render() {
    return (
      <div>
        <Header/>
        <DLCList/>
      </div>
    );
  }
}

export default ResultsPage;
