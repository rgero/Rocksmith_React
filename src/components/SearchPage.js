import React from 'react';
import {connect} from 'react-redux';
//Components
import Header from './Header';
import SearchHeader from './SearchHeader';
import { SearchForm } from './SearchForm';

class SearchPage extends React.Component {

  constructor(props){
    super(props);
    this.onSubmit = this.onSubmit.bind(this);

  }

  onSubmit(query){
    this.props.onSubmit(query);
    this.props.history.push('/results')
  }

  render() {
    return (
      <div>
        <Header/>
        <SearchHeader/>
        <SearchForm onSubmit={this.onSubmit}/>
      </div>
    );
  }
}

export default connect(undefined, undefined)(SearchPage);
