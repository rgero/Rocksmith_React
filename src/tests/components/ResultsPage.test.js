// react-test-renderer
import React from 'react'
import {shallow} from 'enzyme'
import {ResultsPage} from '../../components/ResultsPage';

import dlcData from '../fixtures/dlc';

test("ResultsPage Rendering Test - No Data Rendering", ()=>{
    const wrapper = shallow(<ResultsPage/>);
    expect(wrapper).toMatchSnapshot();
})

test("ResultsPage Rendering Test - Data Rendering", ()=>{
    const wrapper = shallow(<ResultsPage dlc={dlcData}/>);
    expect(wrapper).toMatchSnapshot();
})

