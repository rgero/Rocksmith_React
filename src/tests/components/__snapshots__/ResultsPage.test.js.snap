// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsPage Rendering Test - Data Rendering 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsPage
    dlc={
      Array [
        Object {
          "artist": "Amon Amarth",
          "bassTuning": "E Standard",
          "leadTuning": "Custom Tuning (AEADF#B)",
          "name": "Guardians of Asgaard",
          "rhythmTuning": "Custom Tuning (AEADF#B)",
        },
        Object {
          "artist": "Amon Amarth",
          "bassTuning": "E Standard",
          "leadTuning": "Custom Tuning (B Standard)",
          "name": "The Pursuit of VIkings",
          "rhythmTuning": "Custom Tuning (B Standard)",
        },
        Object {
          "artist": "Amon Amarth",
          "bassTuning": "E Standard",
          "leadTuning": "Custom Tuning (B Standard)",
          "name": "War Of The Gods",
          "rhythmTuning": "Custom Tuning (B Standard)",
        },
        Object {
          "artist": "Amon Amarth",
          "bassTuning": "E Standard",
          "leadTuning": "Custom Tuning (B Standard)",
          "name": "Twilight Of The Thunder God",
          "rhythmTuning": "Custom Tuning (B Standard)",
        },
        Object {
          "artist": "Amon Amarth",
          "bassTuning": "E Standard",
          "leadTuning": "Custom Tuning (B Standard)",
          "name": "Death In Fire",
          "rhythmTuning": "Custom Tuning (B Standard)",
        },
        Object {
          "artist": "Dethklok",
          "bassTuning": "C Standard",
          "leadTuning": "C Standard",
          "name": "Bloodlines",
          "rhythmTuning": "C Standard",
        },
        Object {
          "artist": "Dethklok",
          "bassTuning": "C Standard",
          "leadTuning": "C Standard",
          "name": "Murmaider",
          "rhythmTuning": "C Standard",
        },
        Object {
          "artist": "Dethklok",
          "bassTuning": "C Standard",
          "leadTuning": "C Standard",
          "name": "Black Fire Upon Us",
          "rhythmTuning": "C Standard",
        },
        Object {
          "artist": "Dethklok",
          "bassTuning": "C Standard",
          "leadTuning": "C Standard",
          "name": "Go Into The Water",
          "rhythmTuning": "C Standard",
        },
        Object {
          "artist": "Dethklok",
          "bassTuning": "C Standard",
          "leadTuning": "C Standard",
          "name": "Awaken",
          "rhythmTuning": "C Standard",
        },
        Object {
          "artist": "Dethklok",
          "bassTuning": "C Standard",
          "leadTuning": "C Standard",
          "name": "Thunderhorse",
          "rhythmTuning": "C Standard",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <Connect(DLCList) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <Connect(DLCList) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`ResultsPage Rendering Test - No Data Rendering 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <Connect(DLCList) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <Connect(DLCList) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
